data/mapping/mapping.json

```diff
@@ -4,31 +4,25 @@
       "analyzer":{
         "gp_name_analyzer":{
           "type":"custom",
+          "tokenizer":"name_tokenizer",
+          "filter":[ "lowercase", "gp_title_stop" ]
+        },
+        "surgery_name_analyzer":{
+          "type":"custom",
           "tokenizer":"standard",
-          "filter":[
-            "lowercase",
-            "gp_stop"
-          ]
+          "filter":[ "lowercase" ]
+        }
+      },
+      "tokenizer": {
+        "name_tokenizer": {
+          "type": "pattern",
+          "pattern": "[^a-zA-Z']+"
         }
       },
       "filter":{
-        "gp_stop":{
+        "gp_title_stop":{
           "type":"stop",
-          "stopwords": [ 
-            "and",
-            "the",
-            "&",
-            "doctor",
-            "doctors",
-            "dr",
-            "drs",
-            "partners",
-            "practice",
-            "surgery",
-            "medical",
-            "health",
-            "centre",
-            "center" ]
+          "stopwords": [ "doctor", "doctors", "dr", "drs" ]
```

Comment:
I think your grep needs to be case insensitive. `jq '.[].doctors' data/input/gp-data-merged.json | grep -i 'Doctor' ` gives me matches.

ES removes full stops and commas as well as a load of non-alphanumeric characters (~,@,etc) so the gp_name_tokenizer will split 'dr. andrew smith' into ['andrew', 'smith'].

ES has a cool _analyze API to show what the ouput for a given input would be:
```
curl -XGET 'localhost:9200/profiles/_analyze?pretty' -d'
{
  "analyzer" : "gp_name_analyzer",
  "text" : "Dr. Andrew Smith"
}'
```